custom_components/stiebel_eltron_isg/wpm_coordinator.py

```diff
@@ -121,21 +158,23 @@ def read_modbus_data(self) -> dict:
         """Read the ISG data through modbus."""
         result = {
             **self.read_modbus_energy(),
-            **self.read_modbus_system_state(),
+            **self.read_modbus_system_state1(),
+            **self.read_modbus_system_state2(),
             **self.read_modbus_system_values(),
             **self.read_modbus_system_paramter(),
             **self.read_modbus_sg_ready(),
         }
         return result
 
-    def read_modbus_system_state(self) -> dict:
+    def read_modbus_system_state1(self) -> dict:
         """Read the system state values from the ISG."""
         result = {}
-        inverter_data = self.read_input_registers(slave=1, address=2500, count=47)
+        inverter_data = self.read_input_registers(slave=1, address=2500, count=23)
```

Comment:
I propose not to read twice on the modbus since this reduces the performance. When you need to split the function, I propose to do that after this line

Reply:
Unable to fix in a proper way. I accept suggestions.